os_net_config/impl_ifcfg.py

```diff
@@ -2444,3 +2444,144 @@ def purge(self, iface_name):
                 logger.info(
                     "%s: Device is not managed by ifcfg provider", iface_name
                 )
+
+    def nm_unmanage_devices(self, remove_device_list):
+        """Unmanage a list of devices using nmstate provider.
+
+        """
+        reload_required = False
+        for device in remove_device_list:
+            if (device.provider_data is not None and
+                    device.provider_data.nm_controlled):
+                reload_required = True
+                changed_nm = False
+                logger.info("%s: File %s is changed to NM_CONTROLLED=NO",
+                            device.dev_name, device.provider_data.ifcfg_file)
+                # Change NM_CONTROLLED= in the ifcfg file to NO
+                with open(device.provider_data.ifcfg_file, 'r') as f:
+                    lines = f.readlines()
+                with open(device.provider_data.ifcfg_file, 'w') as f:
+                    for line in lines:
+                        if line.startswith('NM_CONTROLLED='):
+                            f.write('NM_CONTROLLED=NO\n')
+                            changed_nm = True
+                        else:
+                            f.write(line)
+                    if not changed_nm:
+                        f.write('NM_CONTROLLED=NO\n')
+        if reload_required:
+            self.execute("Reloading network", "nmcli", "connection", "reload")
+
+    def remove_devices(self, remove_device_list):
+        """Remove a list of devices using ordered processing.
+
+        This method processes RemoveNetDevice objects in the specified order:
+        ovs_dpdk_port -> ovs_dpdk_bond -> ovs_bond -> vlan -> interface ->
+        sriov_vf -> ovs_bridge -> ovs_user_bridge -> linux_bond -> sriov_pf
+
+        :param remove_device_list: List of RemoveNetDevice objects
+        :type remove_device_list: list[RemoveNetDevice]
+        """
+        if not remove_device_list:
+            logger.debug("remove_devices: No devices to remove")
+            return
+
+        common.print_config(remove_device_list,
+                            msg="removing with ifcfg provider")
+
+        self.nm_unmanage_devices(remove_device_list)
+
+        # Check if we need to backup files for certain device types
+        backup_required_types = ['ovs_dpdk_port', 'ovs_dpdk_bond',
+                                 'sriov_vf', 'sriov_pf']
+        needs_backup = any(device.dev_type in backup_required_types
+                           for device in remove_device_list)
+        if needs_backup:
+            logger.info("Backing up configuration files before device removal")
+            if not self.noop:
+                # Create a backup folder with the current timestamp
+                backup_path = os.path.join(PURGE_IFCFG_FILES,
+                                           common.get_timestamp())
+                os.makedirs(backup_path, exist_ok=True)
+                utils.backup_map_files(backup_path)
```

Comment:
Along with this we could back up the ifcfg files (which are listed in remove_config) ?